types/api/order.d.ts

```diff
@@ -0,0 +1,178 @@
+import { Part } from "../base";
+
+import { Attachment } from "./attachment";
+import { Gender } from "./gender";
+
+export type Order = {
+    id: number;
+    number: number;
+    brand_t?: string;
+    retailer_t?: string;
+    vendor_t?: string;
+    key?: number;
+    status: OrderStatus;
+    status_index: number;
+    production_status?: OrderProductionStatus;
+    production?: string;
+    gender: Gender;
+    currency?: string;
+    country?: string;
+    price?: number;
+    brand: string;
+    shoe: string;
+    variant?: string;
+    shoe_query?: string;
+    shoe_url: string;
+    image_url?: string;
+    minify_hash?: string;
+    image_hash?: string;
+    image_o_hash?: string;
+    top_hash?: string;
+    top_o_hash?: string;
+    bottom_hash?: string;
+    bottom_o_hash?: string;
+    back_hash?: string;
+    side_hash?: string;
+    structure: string;
+    ff_id?: number;
+    ff_shoe_id?: number;
+    ff_order_id?: string;
+    delivery_time?: number;
+    tracking_number?: string;
+    tracking_url?: string;
+    report_url_e?: string;
+    notes?: string[];
+    prices?: {
+        components: Record<string, ComponentPrice>;
+        total: TotalPrice;
+    };
+    states?: OrderState[];
+    attachments?: Attachment[];
+    account?: string;
+    customer?: string;
+    ambassador?: string;
+    store?: string;
+    number_s?: string;
+    channel?: string;
+    delivery_d?: number;
+    scale?: string;
+    size_scaled?: number;
+    details?: OrderDetails;
+};

```

Comment:
@gcandal We should make these types open so that if new fields exist they don't break, how can we do that?

Reply:
We could do so as such:

```ts
type A = {
    a?: number;
    b?: string;
    [x: string]: any;
}

const obj: A = {
    a: 4,
    b: "3",
    c: 5
}
```

Reply:
But then we are not fully taking advantage of the type system (I think). Imagining the `Part` type:

```ts
export type Part = {
    name: string;
    material: string;
    color: string;
    hidden?: boolean;
    optional?: boolean;
};
```
The  following would be allowed:
```ts
const part: Part = {
    name: "laces",
    material: "crocodile",
    colour: "yellow" // notice the key "colour" is misspelled
};
```

Reply:
Ok let's do it for any type coming from Appier Models

Reply:
I agree that if we're using types might as well be as strict and complete as possible. In the long run we save time in bugs prevented by the type system and also get really nice "documentation" because parameters like `options` become clearer.

Also, the fix I presented may not be the best but from some quick research of the web and TS documentation it looks like the only one.

Reply:
@joamag do we keep the type open anyway?

Reply:
Let's reverse the change, as unpacking a value from JS with more fields does not generate a runtime error

Reply:
![image](https://user-images.githubusercontent.com/3048875/106491262-97468380-64ae-11eb-9895-efd5b4539ff1.png)